### PR TITLE
Migrate `tests/run-make/prefer-dylib` to `rmake.rs`

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of https://github.com/rust-lang/rust/issues/121876.

r? @jieyouxu

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Ah new API. This one looks better indeed!

File: tests/run-make/prefer-dylib/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+use std::fs::remove_file;
+use std::process::Command;
+
+fn main() {
+    rustc().input("bar.rs").crate_type("dylib").crate_type("rlib").arg("-Cprefer-dynamic").run();
+    rustc().input("foo.rs").arg("-Cprefer-dynamic").run();
+
+    run("foo");
+
+    read_dir(cwd(), |path| {
```
Comment (Kobzol) [inline tests/run-make/prefer-dylib/rmake.rs:13]: I know that this was in the original Makefile, but I think that we can improve it to this:
```rust
remove_file(dynamic_lib_name("bar")).unwrap();
```
To make it easier to understand what's going on. Removing the dynamic library is enough to keep the test working.

**Review by jieyouxu — APPROVED** [review]
Thanks! r=me after applying Kobzol's suggestion.

**Review by jieyouxu — APPROVED** [review]
Thanks for the changes! r=me after CI is green.

**Pull request merged**